cpp/include/cudf/io/detail/utils.hpp

```diff
@@ -25,5 +26,15 @@ namespace io::detail {
  */
 enum class single_write_mode : bool { YES, NO };
 
+template <typename T>
+constexpr bool is_handled()
+{
+  // Note: the case (not std::is_same_v<T, bool>)
+  // is already covered by is_integral
+  return std::is_same_v<T, cudf::string_view> || std::is_integral_v<T> ||
+         std::is_floating_point_v<T> || cudf::is_fixed_point<T>() || cudf::is_timestamp<T>() ||
+         cudf::is_duration<T>();
+}
```

Comment:
*nit*: Can we at least add documentation for this function? `is_handled` is quite a generic name. (Previously it was a static member function of a `column_to_strings_fn` struct, so `column_to_strings_fn::is_handled<type>()` is more self-explanatory).

Perhaps:

```
@brief Is conversion from a column `T` to a string column is supported for IO purposes
@tparam T Column type to check.
@return true if conversion is supported, false otherwise.
```
?

We could perhaps call this function `is_convertible_to_string_column<T>()`?